frameworks/hdfs/tests/test_sanity.py

```diff
@@ -148,10 +148,23 @@ def test_kill_data_node():
 @pytest.mark.sanity
 @pytest.mark.recovery
 def test_kill_scheduler():
+    foldered_name = sdk_utils.get_foldered_name(config.SERVICE_NAME)
+    scheduler_name = sdk_utils.get_scheduler_name(foldered_name)
+    scheduler_id = sdk_tasks.get_task_ids(foldered_name, scheduler_name)
```

Comment:
Looking at the test output I see:
```
[2018-08-15 11:29:22,8
02|sdk_tasks-check_task_relaunched(210)|INFO]: Checking (task_name:hdfs.integration.test) (old_task_id:[]) with (ensure_new_task_not_completed:True) is relaunched
[2018-08-15 11:29:22,804|sdk_cmd-cluster_request(82)|INFO]: (HTTP GET) /mesos/tasks
[2018-08-15 11:29:22,874|sdk_cmd-fn(99)|INFO]: Got 200 for GET /mesos/tasks
[2018-08-15 11:29:22,879|sdk_cmd-cluster_request(82)|INFO]: (HTTP GET) /mesos/slaves
[2018-08-15 11:29:22,899|sdk_cmd-fn(99)|INFO]: Got 200 for GET /mesos/slaves
[2018-08-15 11:29:22,916|sdk_tasks-get_summary(149)|INFO]: Task summary (with_completed=True) (task_name=[hdfs.integration.test]):
- Task[name="hdfs.integration.test"	state=TASK_RUNNING	id=test_integration_hdfs.723eed03-a07e-11e8-8d87-0209f1235b3f	host=10.0.3.178	framework_id=74e0d766-a764-4b55-88a2-8d8a24e4c716-0001	agent=74e0d766-a764-4b55-88a2-8d8a24e4c716-S7]
- Task[name="hdfs.integration.test"	state=TASK_FAILED	id=test_integration_hdfs.213d89e2-a07c-11e8-8d87-0209f1235b3f	host=10.0.3.178	framework_id=74e0d766-a764-4b55-88a2-8d8a24e4c716-0001	agent=74e0d766-a764-4b55-88a2-8d8a24e4c716-S7]
- Task[name="hdfs.integration.test"	state=TASK_KILLED	id=test_integration_hdfs.015613f1-a07b-11e8-8d87-0209f1235b3f	host=10.0.3.235	framework_id=74e0d766-a764-4b55-88a2-8d8a24e4c716-0001	agent=74e0d766-a764-4b55-88a2-8d8a24e4c716-S9]
- Task[name="hdfs.integration.test"	state=TASK_KILLED	id=test_integration_hdfs.66228ccc-a078-11e8-8d87-0209f1235b3f	host=10.0.1.127	framework_id=74e0d766-a764-4b55-88a2-8d8a24e4c716-0001	agent=74e0d766-a764-4b55-88a2-8d8a24e4c716-S6]
- Task[name="hdfs.integration.test"	state=TASK_KILLED	id=test_integration_hdfs.9c4f7c6a-a076-11e8-8d87-0209f1235b3f	host=10.0.2.13	framework_id=74e0d766-a764-4b55-88a2-8d8a24e4c716-0001	agent=74e0d766-a764-4b55-88a2-8d8a24e4c716-S5]
```
Note the first line which contains: `(old_task_id:[])`. This seems to indicate that `scheduler_id` is not set as expected. Note, the Marathon output earlier in the log shows both the previous and the current task ID for the scheduler.

Looking at the code for `check_task_relaunched`, it seems as if `old_task_id` is supposed to be a string, and not a list that is being returned by `get_task_ids`.

So, we need:
* To return the valid scheduler ID (or just get it from the marathon app state) from `get_task_ids`
* Pass a **single** string to `check_task_relaunched `.

Reply:
May have ended up fixing this test via the shakedown removal PR? May be able to do something like this: https://github.com/mesosphere/dcos-commons/pull/2616/files#diff-a0ec6572c50515f49ce94fae88201609R146